File: frontend/eop_online/src/domain/match/types/Card.ts
export type Card = {
    cardNumber: number
    value: string
    suit: Suit
    text: string
    example: string
    mitigation: string
}

export type Suit =
    | "Spoofing"
    | "Tampering"
    | "Repudiation"
    | "InformationDisclosure"
    | "DenialOfService"
    | "ElevationOfPrivilege"

File: src/main/scala/com/github/pawelj_pl/eoponline/match/Card.scala
package com.github.pawelj_pl.eoponline.`match`

import io.circe.Codec
import io.circe.generic.semiauto.deriveCodec

final case class Card(cardNumber: Int, value: Value, suit: Suit, text: String, example: String, mitigation: String)

object Card {

  implicit val codec: Codec[Card] = deriveCodec

}
